service/resource/deployment/deployment.go

```diff
@@ -4,38 +4,75 @@ import (
 	"fmt"
 
 	"github.com/giantswarm/azuretpr"
+	"github.com/giantswarm/microerror"
 
 	"github.com/giantswarm/azure-operator/service/key"
 )
 
 const (
-	clusterIDParameterName     = "clusterID"
-	clusterSetupDeploymentName = "cluster-setup"
-	clusterSetupTemplate       = "cluster_setup.json"
-	templateBaseURI            = "https://raw.githubusercontent.com/giantswarm/azure-operator/master/service/arm_templates"
-	templateVersion            = "1.0.0.0"
+	mainDeploymentName          = "cluster-main-template"
+	templatesBaseURI            = "https://raw.githubusercontent.com/giantswarm/azure-operator/%s/service/arm_templates/"
+	templateVersion             = "1.0.0.0"
+	mainTemplate                = "main.json"
+	clusterSetupTemplate        = "cluster_setup.json"
+	securityGroupsSetupTemplate = "security_groups_setup.json"
+	virtualNetworkSetupTemplate = "virtual_network_setup.json"
 )
 
 func getDeploymentNames() []string {
 	return []string{
-		clusterSetupDeploymentName,
+		mainDeploymentName,
 	}
 }
 
-func newClusterSetupDeployment(cluster azuretpr.CustomObject) (Deployment, error) {
+func (r Resource) newMainDeployment(cluster azuretpr.CustomObject) (Deployment, error) {
```

Comment:
Here the main template deployment is created. The child templates are called from this template via template links.